app/models/profile.rb

```diff
@@ -22,9 +24,29 @@ def decrypt_pii(user_access_key)
     Pii::Attributes.new_from_encrypted(encrypted_pii, user_access_key)
   end
 
+  def recover_pii(recovery_code)
+    raise 'invalid recovery code' unless recovery_code_generator.verify(recovery_code)
+    uak = recovery_code_generator.user_access_key
+    Pii::Attributes.new_from_encrypted(encrypted_pii_recovery, uak)
+  end
+
   def encrypt_pii(user_access_key, pii)
     ssn = pii.ssn
     self.ssn_signature = Pii::Fingerprinter.fingerprint(ssn) if ssn
     self.encrypted_pii = pii.encrypted(user_access_key)
+    recovery = generate_recovery_code
```

Comment:
what about being more descriptive with:

```
recovery_code, user_access_key = generate_recovery_code
```

so you don't have to use `recovery[1]` and `recovery[0]` below?